Fix node installation for 32 bit systems

ankita240796 opened the pull request:
## Explanation
Fixes node installation for 32 bit systems.

## Checklist
- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python -m scripts.pre_commit_linter` and `python -m scripts.run_frontend_tests`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "PROJECT: ..." label (Please add this label for the first-pass review of the PR).
- [x] The PR has an appropriate "CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

ankita240796 committed:
Fix urls for 32 bit system

ankita240796 commented:
@DubeySandeep @seanlip can anyone of you please test this on your system since I was testing this on my work laptop and the command did not run completely due to some permission issues.

To test:

- Add `node_file_name = 'node-v10.15.3'` [here](https://github.com/oppia/oppia/pull/7882/files#diff-469c5b07dc3b88dda7d15b7ec4e049cfR137).
- Run `python -m scripts.setup`

Thanks!

Jethet committed:
Fix part of #6975 Apply the new naming convention to remaining directories (#7844)

* Apply new naming convention to folders /domain/subtopic_viewer to /domain/utilities

* Address reviewer remarks

* Last changes requested by reviewer

* Address comments reviewer

seanlip reviewed (changes requested):
I don't have a 32-bit system, sorry. I strongly recommend that someone actually does test this on a 32-bit system before further review is undertaken, though (ideally, they are assigned as the main reviewers).

Perhaps ask on oppia-dev@? I'm concerned about possible side effects of the new ./configure / make calls -- e.g. we don't want to end up with a bunch of spurious files sitting around somewhere as artifacts of this process.

kevinlee12 commented:
It's unlikely that we'll have developers with 32 bit systems, so I don't think it's worth waiting too long. My recommendation is to install a 32 bit OS on a VM and test it out there. Unfortunately I don't have the bandwidth to do that, but that test will suffice without depending on others to test it.

ankita240796 commented:
Hi @seanlip @kevinlee12, I will test this on a VM and assign you for review once I have done that! Thanks!

jamesxu0 committed:
changed margins (#7883)

seanlip committed:
Routine update of translations. (#7887)

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Add Hungarian to list of supported languages.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix broken tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix translation error.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix translation error.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix errors in translations found by backend tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix broken tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix errors in translations.

* Fix a Python linting issue.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix spelling error in description

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix backend tests.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix French translation.

* Localisation updates from https://translatewiki.net.

* Fix French translation.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Localisation updates from https://translatewiki.net.

* Fix HTML errors.

vojtechjelinek committed:
Wipeout 2.7: Reindex all children of BaseSnapshotMetadataModel (#7874)

* Add deletion policies for topic models

* Add deletion policies for skill models

* Add deletion policies for question models

* Add deletion policies for email models

* Add deletion policies for feedback models

* Add test deletion policies

* Add get_deletion_policy to models

* Add skip to the get_deletion_policy test

* Fix lint errors

* Add test

* Modify test

* Change comments

* Add get_deltion_policy to other models

* Add recommendations tests

* Reorder stats tests

* Add tests for get_deletion_policy in stats

* Modify test

* Fix lint issues

* Change policy for question-skill link

* Edit and fix tests

* Add has_reference_to_user_id to some models

* Fix lint

* Assert review comments

* Assert review comments

* Fix linter errors

* Fix comments

* Add has_reference_to_user_id for activity models

* Add has_reference_to_user_id for audit models

* Add has_reference_to_user_id for classifier models

* Add has_reference_to_user_id for config models

* Add has_reference_to_user_id for email models

* Add has_reference_to_user_id for feedback models

* Modify meta tests

* Fix lint tests

* Correct has_reference_to_user_id methods

* Add has_reference_to_user_id to opportunity models

* Add has_reference_to_user_id to job models

* Add has_reference_to_user_id to file models

* Add has_reference_to_user_id to statistics models

* Fix backend tests

* Add has_reference_to_user_id to suggestion models

* Add has_reference_to_user_is to collection models

* Add has_reference_to_user_is to exploration models

* Add has_reference_to_user_is to question models

* Add has_reference_to_user_is to skill models

* Fix formatting

* Move has_reference_to_user_id to BaseCommitLogEntryModel

* Add has_reference_to_user_is to story models

* Fix typos

* Add has_reference_to_user_is to topic models

* Fix typos

* Remove has_reference_to_user_id

* Improve according to previous comments

* Improve according to previous comments

* Fix backend tests

* Assert comments

* Modify has_reference_to_user_id to classmethods

* Fix coverage

* Add TODO comment

* Add one-off job fro reindexing activities

* Add migration job and tests

* Address comments

* Improve backend tests

* Address comments

* Change remaining staticmethods to classmethods

* Fix formatting

* Fix formatting

* Add indexed to base model

* Address comments

* Address comment

* Rename exploration_models to exp_models

Commit:
fix inprog issue (#7885)

nadintamer committed:
Fix part of #7427: Add valid args docstring for story_domain_test.py (#7832)

* Add args to _assert_validation_error docstring

* Add args to _assert_valid_story_id docstring

* Indent args

* Wrap long lines

* Change string to str

* Change string to str

aks681 committed:
Changes to questions tab and rubrics editor (#7846)

* Changes to questions tab and rubrics editor

* Minor change

* review changes and e2e test fix

* fixed tests and made review changes

cuichenli committed:
Fixes part of #6240:  Mark correct answers after enabling correctness feedback (#7764)

* Draft version

* Draft version for mark correctness  e2e test

* Another test

* Add e2e test for enable correctness feedback.

Draft

* Fix addResponse.apply

* Add e2e test for exploration players with correctness feedback enabled

* Add more conditions and restructure

* Change it statements

* Move enable correctness e2e test into core editor features

* Rename elements and add some logic

* Fix lint issues

* Remove mistakenly added file

* Reorganize after each and before each

* Remove redundtant suite

* Rename applyCommonSetitngs to enableMarkCorrectness

* Fix issues

- Remove redundant spaces
- Remove redundant import
- Reorder import statements
- Rename utility function

vojtechjelinek committed:
Wipeout 3.4: Start using GAE user ID (#7823)

* Fix link sharing dialog

* Generate new user_id

* Fix comment

* Add prefix to user_id

* Fix system admin user

* Fix test utils to work with separate GAE id

* Stop evaluating user ids

* Fix tests

* Assign gae user id to user id

* Fix lint

* Fix backend tests

* Assert comments

* Fix backend tests

* Fix backend tests

* Fix backend tests

* Fix backend tests

* Change gae_user_id to gae_id

* Add backend tests

* Improve tests

* Address comments

Commit:
Fix #6909: Add a polyfill for SVGElement.prototype.outerHTML (#7886)

* Add SVGElement.prototype.outerHTML polyfill for IE

* add comment

* fix indentation issues

Hudda committed:
Fix #7831: Added lint check for excessive new lines between function definitions (#7881)

* Added lint check for excessive new lines between function definations

* Added lint checks for excessive blank lines between function definitions

* Removed extra newline from core.controllers.base

kevinlee12 committed:
Fix #7890: Add more explicit codeowners fix (#7895)

* Make CODEOWNERs file message more explicit

* Fix lint error

* Add messages to fix lint errors

* Fix indent errors

* Fix parenthesis issue

* Fix linter errors

* Fix linter errors

aks681 committed:
Removed the concept of "private skills" (#7891)

Commit:
Fix #7794: Add warning when MIDI is unsupported in the browser (#7840)

aks681 committed:
Added coverage for skill_editor.py (#7905)

Commit:
Fix #7529: Adds a strict check for Outcome object before executing hasNonemptyFeedback (#7856)

* Execute outcome.hasNonemptyFeedback only if outcome is present

* remove extra semicolon

* Make outcome check strict

* handle null condition

lorryaze committed:
Fix part of #7450: Modify private methods calls in GenerateHashTests to public calls (#7860)

* changing a private test call from the generate_hash method to a public call

* fix tests of _generate_hash

* fix pylint error

* fixing generate_hash test

xlogix committed:
Fix part of #6975: Rename services directory files according to new naming conventions (#7896)

* Fix part of #6975, rename service directory

* Fix part of #6975, rename service directory #2

* Fix part of #6975, rename service directory, linter fix

* Fix part of #6975, rename service directory, linter fix #2

* Modify codeowners file to include new names

* Update codeowners file

* Fix import which prevents contunous integration test to pass

* Merge recent changes

* fix newline

xlogix committed:
Fix #6738: Change direct usage of window.location (#7879)

ankita240796 committed:
Release Automation [Milestone 4]: Add pre commit tests in deploy, automate hotfix branch creation (#7790)

* Add tests for deploy script

* Make coverage 100%

* Fix name

* Add check for configs

* Add newline

* Fix name

* Add tests for clean.py

* Add tests for gcloud adapter

* Add test for install third party

* Add tests for install third party libs

* Add test for pre push hook script

* Add test for pre commit hook script

* Add try finally

* Add test for setup scripts

* Add test for setup gae script

* Add extra checks in deploy script

* Fix linter

* Add pre commit test for config updates

* Make coverage 100%

* Add test for last commit message

* Add automation for cutting hotfix branch

* Address review comments

* Address review comments

* Make coverage 100%

* Update test

* Address review comments

vinitamurthi committed:
Add more interaction types to the question editor and fix question mode detection in outcome editor (#7894)

* Convert strings to i18n strings

* Merge conflict fixes

* Add more interaction options and fix the question mode recognition in outcome editor

* Lint fix

* Review fixes

aks681 committed:
Split flag for new structure viewers (#7898)

* Split flag

* added comment

* fixed test

bansalnitish committed:
Upgrades AlertsService and NormalizeWhiteSpaceFilter to Angular (#7777)

* upgrade

* Fixes the opening of splash page

* Adds correct way to inject

* Upgrades AlertsService to Angular

* Upgrades NormalizeWhitespace filter

* Resolves lint errors

* Adds optional arguments in loggerservice

* Addresses Review comments

* Addresses review comments

* Fixes frontend tests

* Addresses comments

* Addressed comments

Showtim3 committed:
Migrating dateTimeFormatService and format-timer.pipe (#7827)

* migrating dateTimeFormatService and format-timer.pipe

* migrating DateTimeSpec file

* fixed styling

* fixed order

* merging develop

* fixing transform

* changing timer to time

* moving filter

* fixing types

seanlip committed:
Switch download source for CKEditor libs to GitHub for better stability. (#7920)

* Switch download source for CKEditor libs to GitHub for better stability.

* Update skip list in app_dev.yaml

* Reinstate angular-drag-and-drop-lists since it is referenced directly in header_js_libs.html

Showtim3 committed:
Migration url service (#7834)

* migrating UrlService

* fixing realtive imports

* improving the styling

* fixing tests

* removing Error from return type, fixing imports

* Added a comment explaining return type

* Making the Pr url more specific

* fixing imports

aks681 committed:
Flipped flag (#7889)

aks681 committed:
Added fields to skill editor (#7909)

* updated skill_domain and skill_services

* minor change to test file

* Added must_be_addressed field to misconception editor

* Added prerequisite skills to editor

* lint

* Changed UI

* Changed var name and added comments

* changed comments

* fixed tests

* fixed test

* fixed test

* fixed test

DubeySandeep committed:
Remove DubeySandeep from codeowner for various miscellaneous projects. (#7914)

* Removed DubeySandeep from various projects.

* Changes as per review comments

ankita240796 committed:
Release Automation [Milestone 6]: Automate backup restoration for migrations on backup server (#7916)

* Automate backup restoration

* Address review comments

* Use dummy names

* Improve dummy names

* Fix comment

ankita240796 committed:
Release Automation [Milestone 5]: Add checks to ensure that travis and circleci are run before prod deployment, changelog generation for hotfixes (#7912)

* Add tests for deploy script

* Make coverage 100%

* Fix name

* Add check for configs

* Add newline

* Fix name

* Add tests for clean.py

* Add tests for gcloud adapter

* Add test for install third party

* Add tests for install third party libs

* Add test for pre push hook script

* Add test for pre commit hook script

* Add try finally

* Add test for setup scripts

* Add test for setup gae script

* Add extra checks in deploy script

* Fix linter

* Add pre commit test for config updates

* Make coverage 100%

* Add test for last commit message

* Add automation for cutting hotfix branch

* Address review comments

* Address review comments

* Make coverage 100%

* Update test

* Address review comments

* Changelog generation for hotfixes

* Add tests to ensure that tests are run before deploying

* Add newline

* Add new lines, update common test

* Fix version

* Update checking of travis and circleci

* Fix changelog creation

* Use constants

Showtim3 committed:
Migration extension tag assembler serivce (#7805)

* upgraing htmlEscaper, ExtensionTagAssembler

* fixed linting

* adding types

* fixing typescript

* remvoing unncessary ts-ignore

* fixing imports

* fixing import

* merging changes

* adding testbed to htmlEscaperService

* fixing tests

* fixing imports and styling

* adding todo for any

* adding new line

* fixing imports

* breaking after the (

* fixing imports

* adding codeowner file

* removing codeowner

vojtechjelinek committed:
Wipeout 2.8: Add has_reference_to_user_id to remainig models (#7893)

* Add deletion policies for topic models

* Add deletion policies for skill models

* Add deletion policies for question models

* Add deletion policies for email models

* Add deletion policies for feedback models

* Add test deletion policies

* Add get_deletion_policy to models

* Add skip to the get_deletion_policy test

* Fix lint errors

* Add test

* Modify test

* Change comments

* Add get_deltion_policy to other models

* Add recommendations tests

* Reorder stats tests

* Add tests for get_deletion_policy in stats

* Modify test

* Fix lint issues

* Change policy for question-skill link

* Edit and fix tests

* Add has_reference_to_user_id to some models

* Fix lint

* Assert review comments

* Assert review comments

* Fix linter errors

* Fix comments

* Add has_reference_to_user_id for activity models

* Add has_reference_to_user_id for audit models

* Add has_reference_to_user_id for classifier models

* Add has_reference_to_user_id for config models

* Add has_reference_to_user_id for email models

* Add has_reference_to_user_id for feedback models

* Modify meta tests

* Fix lint tests

* Correct has_reference_to_user_id methods

* Add has_reference_to_user_id to opportunity models

* Add has_reference_to_user_id to job models

* Add has_reference_to_user_id to file models

* Add has_reference_to_user_id to statistics models

* Fix backend tests

* Add has_reference_to_user_id to suggestion models

* Add has_reference_to_user_is to collection models

* Add has_reference_to_user_is to exploration models

* Add has_reference_to_user_is to question models

* Add has_reference_to_user_is to skill models

* Fix formatting

* Move has_reference_to_user_id to BaseCommitLogEntryModel

* Add has_reference_to_user_is to story models

* Fix typos

* Add has_reference_to_user_is to topic models

* Fix typos

* Remove has_reference_to_user_id

* Improve according to previous comments

* Improve according to previous comments

* Fix backend tests

* Assert comments

* Modify has_reference_to_user_id to classmethods

* Fix coverage

* Add TODO comment

* Add one-off job fro reindexing activities

* Add migration job and tests

* Address comments

* Improve backend tests

* Address comments

* Change remaining staticmethods to classmethods

* Fix formatting

* Fix formatting

* Add indexed to base model

* Address comments

* Address comment

* Rename exploration_models to exp_models

* Remove ReviewerRotationTrackingModel since it is no longer used

* Add migration for mail models

* Remove all code that used ReviewerRotationTrackingModel

* Remove has_reference_to_user_id from TopicSimilaritiesModel

* Add has_reference_to_user_id to QuestionSkillLinkModel

* Add has_reference_to_user_id to StorySummaryModel

* Add has_reference_to_user_id to SkillSummaryModel

* Add has_reference_to_user_id to TopicSummaryModel

* Add has_reference_to_user_id to email models

* Remvoe skip from has_reference_to_user_id tests

* Fix backend tests

* Address comments

vinitamurthi committed:
Question editor fixes (#7910)

* Convert strings to i18n strings

* Merge conflict fixes

* Left align question content, stop multiple modals from opening

* Expand the width of oppia tells the learner

* Left align content

* Add wording to story node editor

* lint fixes

* Change the warning dialog for question editor

* lint fixes

* change test

* review fixes

* lint fixes

* Remove extra space

* Fix test

DubeySandeep committed:
Resolves ckeditor.js 404 issue on the community dashboard page (#7924)

Commit:
Fix part of #6482: button is disable until all requirements are fulfilled - fixed \o/ (#7908)

* issue #6482 fixed - Skill editor, Questions, button Save and Publish Question is disable untill all requirements are fulfilled

* 2 errors that appeared on push fixerd

* max lenght error fixed

* space error fixed

* function name corrected and tooltip added

seanlip committed:
Fix #7867: do client-side redirect for app pages. (#7921)

* Fix #7867: do client-side redirect for app pages.

* Fix typo.

kevinlee12 committed:
Fixes #7647: Subprocess check call conversion (#7930)

* Convert subprocess call to check_call

* Revert conversion

* Remove unneeded travis config value

* Undo check call linter

* Revert check_call in common_test

* Fix linter test

DubeySandeep committed:
Voiceover application storage model and domain class (#7843)

* Created models and domain class.

* Few typo fix.

* Adds accept and reject methods for exploration voiceover application calss.

* Undo changes in email/gae_model

* Adds prod model validator class with test.

* Adds missing test for services.

* Self review changes.

* Changes as per review comments.

* Address review comments.

* Added object validation error test for prod validation.

Showtim3 committed:
Angular Services Migration combined (#7934)

* migration debouncer service

* migrating computeGraphService

* fixing import order

* migrating GenerateContentIdService

* fixing imports

* adding todo for any

* adding todo for any

* adding new lines

* fixing any data types

* removing empty line

* fixing imports

* fixed imports

* Adding args types

* fixing imports

ankita240796 committed:
Update tests according to the changes in #7930 (#7941)

* Remove extra kill cmd

* Update tests

Commit:
Adds a check for errorCallbac (#7927)

Commit:
Fix #7342:  Replace ctrl.stateName with updated state name value (#7907)

* fix interaction undefined and other issues

* Fix lint issues

Showtim3 committed:
Migrating filters (#7935)

* Migrating string-utility-filters

* migrating more string-utility-filters

* fixing tests

* fixing imports

* fixing tests

* fixing tests

* removing unwanted whitespace

* removing whitespace

tyleri committed:
Run install_third_party_libs.py script before running e2e tests (#7918) (#7947)

Commit:
Fix #7926: Add polyfill for navigator.mediaDevices.getUserMedia (#7933)

* Add polyfill for getUserMedia

* suppress typescript compilation error

marianadasilvadev committed:
Fix #6687: Write e2e tests for file upload process (#7839)

kevinlee12 committed:
Resolve audit errors (#7949)

ankita240796 committed:
Revert "Wipeout 3.4: Start using GAE user ID (#7823)" (#7954)

This reverts commit c60361e4ef32f01b0da126c24aba4174b99634f5.

Commit:
Fix #7784: Add type as a second parameter when contructing a blob (#7785)

* Add data type as a second parameter when contructing a blob

* remove if block

* fix spec

nishantwrp committed:
Fix part of #6240: Add e2e tests for testing navigation buttons in an exploration (#7888)

* Add e2e tests for testing navigation buttons in an exploration

* Corrected a typo

* Changed the error message

* Minor Changes

* Some more changes

* Capitalize the describe message

* Move some code from beforeAll() to it

* Add aftereach() block

Commit:
Fix #6158: Log for interpolate url issue (#7952)

* Adds log

* fix lint issues

ankita240796 commented:
Working on it. I will try to update by tomorrow.

bansalnitish committed:
Upgrades NumberWithUnits related services to Angular (#7944)

* Upgrades NumberWithUnit to Angular

* Added NUmberInput to UpgradedServices

* Upgrades NumericWithUnitInputRule to Angular

* Upgrades NumberWithUnitsValidation to Angular

* Fix conflicts

* Fixes travis

* Adds require in directives

* Fixes e2e tests

* Fixes tests

* Resolves conflicts

* Resolves comments

* Address comment

kevinlee12 committed:
Fix skulpt install (#7953)

* Fix skulpt install

* Change alias in travis config

* Fix skulpt install test

vojtechjelinek committed:
Remove fonts directory (#7978)

varun-tandon committed:
Fix part of #5914: Add testing for nonexistent users in export_data (#7942)

* add tests for nonexistent users

* linting

* linting

* assert exception

* resolve linting too long line

* resolve invalid continuation

Hudda committed:
Fix #7868: Added lint check for a newline above args in python doc string (#7929)

* Added lint check for newline above args

* Added more tests

* Added lint check for newline above args

* Split test in modules

kevinlee12 committed:
Resolve comments from PR 7953 (#7979)

tyleri committed:
Addresses docstring issues in one of the files (prod_validation_jobs_one_off.py) from #7427 (#7948)

hamletv committed:
Fix #7849: Add background color when hovering over sign-button (#7917)

* new css rule removes white background (#7849)

* changed location of added css rule to pass linting

* removed dupliate css rule to pass linting

ankita240796 committed:
Fix #7963: Adds margin for logic proof help lines (#7997)

* Add margin

* Add container div

vojtechjelinek committed:
Rename gae_user_id to gae_id (#7998)

Showtim3 committed:
Fixes 7939: Admin navbar misplaced fix (#7940)

* fixing admin navbar

* Removing property from oppia.css

Commit:
minor css fixes (#7986)

U8NWXD committed:
Fix Part of #6240: Subtopic Editor Drag and Drop Skills E2E Test (#7901)

* Install html-dnd for Protractor Drag-Drop Tests

A bug in Selenium (see
https://code.google.com/p/selenium/issues/detail?id=3604) prevents us
from using normal Protractor syntax to simulate drag-and-drop events in
a test. Instead, we can use the html-dnd NPM package to simulate
drag-and-drop events.

Fixes part of #6240.

* Add E2E Test for Dragging Skills To Subtopics

In the topic editor, the user can drag skills to assign them to
subtopics. This commit adds an end-to-end test to check that this
functionality works. Note that because of a bug in Selenium preventing
the use of the normal protractor drag-and-drop commands, this commit
makes use of the `html-dnd` package.

Fixes part of #6240.

* Test Dragging Skill Between Subtopics

Update existing e2e test to also check that skills can be dragged
between subtopics.

Fixes part of #6240.

* E2E Test Drag Skill to Uncategorize

Update E2E test to drag skill from subtopic to the uncategorized skills
area to uncategorize the skill.

Fixes part of #6240.

* Refactor Skill Drag to Subtopic Test HTML Classes

To simplify the E2E test checking the dragging of skills to, between,
and from subtopics, this commit replaces the `xpath('..')` used to get
parent DOM elements with a new `protractor-test-*` HTML class for
subtopic columns.

Fixes part of #6240.

* Rename Dragging Skills E2E Test

Renaming reflects that test now covers dragging skills between and from
subtopics, not just to them.

Fixes part of #6240.

* Create DragDrop Function and Replace const -> var

Two changes:
* To reduce redundant calls to `browser.executeScript(...)`, create a
  dragAndDrop helper function.
* Use `var` instead of `const` for the `html-dnd` code to match file
  conventions.

Fixes part of #6240. Addresses review comments from @Showtim3.

* Use Drag and Drop Helper Function

* Make Drag-Drop Tests Compatible with Others

Because test suites are run on the same instance of the server, the
drag-and-drop test runs on the state left by the previous tests. This
commit updates the test accordingly.

Showtim3 committed:
Migration angular services: changes in human readable (#7945)

* migrating services and exporting interation specs

* migrating IdGenerationService

* adding document

* migrating services

* fixing tests

* adding types

* adding types

* fixing typescript

* fixing lint

* adding types, fixed spacing

* fixing imports

* fixing relative path

* adding return types

* addressing review

aks681 committed:
Fix #7967: ensure chapter editor validation failures are not cut off on right (#8001)

kevinlee12 committed:
Resolve yarn audit (#8006)

Commit:
Fix #7975: 'this' was out of scope in SidebarStatusService (#8011)

* fixes hamburger issue

* fix lint issue

rajitha1998 committed:
fix user profile image issue (#8017)

seanlip committed:
Add more explanation to the top of the run_backend_tests script. (#8012)

DubeySandeep committed:
Fix #7968: Adds missing language code in the exploration opportunity domain object (#7999)

* Adds new language code in the domain object

* Remove unwanted methods

* Address review comments.

Commit:
Fix #7124: add css to preferences-bio textarea (#8010)

* fixes #7124

* fixes #7124

* fixes #7124 (proper code style)

* css added for bio in preferences

* fix extra line and class mentioning

* fixes #7124

* fixes #7124

* fixes #7124 (proper code style)

* css added for bio in preferences

* fix extra line and class mentioning

* changed class name to oppia-bio-border

brianrodri committed:
Fix UpgradedServices to use true dependency management (#8013)

* use getUpgradedServices

* minor fixes

* fix missing callsites

* fix remaining callsites

* move playthrough improvement task upgraded services

* address lint errors

* fix indent error

* better organization

* use block comments

ankita240796 committed:
Fix date, publish button (#7996)

ankita240796 commented:
Closing for a while, will re-open after testing.

Commit:
fix overflow issue (#8003)

Commit:
Fix #7691: Card Basic Setting Alignment Fixed (#7904)

* card basic setting alignment fixed

* (dubeySandeep request) erase extra space in class

Commit:
Fix #8032: Truncate skill name to 35 characters for display (#8033)

* fixes skill name overflow in navbar

* fixes overflow issue in the description; adds 200 char limit to skill name in the modal and editor page;

* fix trailing spaces

* reduce maxlength to 100

DubeySandeep committed:
Services required for voiceover application  (#7857)

* Created models and domain class.

* Few typo fix.

* Changes in services for voiceover application.

* Adds accept and reject methods for exploration voiceover application calss.

* Some minor changes.

* Undo changes in email/gae_model

* Adds prod model validator class with test.

* Adds missing test for services.

* Self review changes.

* Changes as per review comments.

* Fix backend test issues.

* Address review comments.

* 100% test coverage.

* Chnages as per review comments.

* Fix test issue.

* Change hardcoded id.

* Change more hardcoded id.

* Rejectes other applications while accepting one application.

donosco99 committed:
Fixes #7430: Added a directive to prevent title overflow from the exploration info box  (#7985)

* added directive to prevent overflow of title in exploration info box

* added directive to prevent overflow of title in exploration info box

* added directive to prevent overflow of title in exploration info box

vpishuk committed:
Fixes part of #7450: removes all calls to user_services._save_user_settings from user_services_test.py (#7972)

* fix: call to the private method in LastLoginIntegrationTests

The change includes:
* removal of using private method called "_save_user_settings"
* adds "mock_datetime_utcnow(datetime)" method to test_utils;
  the method is useful to mock utcnow to a proper time
  in a context

* style: fixes order of imports inside test_utile.py

* style: removes unused variable

* style: removes unused import

* style: add line-break after opened parenthesis

* style: reduce the length of the line up to 80 symbols

* fix(user_services_test): removes unnecessary change to user_settings before checking initial state

* test(user_services): replaces call to private method in LastExplorationEditedIntegrationTests

* test(user_services): removes redundant update of settings in LastExplorationCreatedIntegrationTests

* test(user_services): remove call to private method in LastExplorationCreatedIntegrationTests

* fix(test_utils): improves mock_datetime_utcnow method

* adds validation for mocked_datetime arg
* adds Docstring

* fix(test): fixes comments in code-review

* fix(test_utils): fixes spelling of datetime

Hudda committed:
Fixes part of #5939: Custom pylint extension for checking division operator (#8034)

* Added lint check for division operator

* Split test case

aks681 committed:
Empty rubric explanation allowed with placeholder (#8042)

* Allowed empty rubrics to be present and added placeholder for the same

* minor fix

* made review change

Commit:
Fixes (partially): oppia can not not run on Windows  (#7971)

* Dirty tweak

* Fix pre_commit_linter

* Fix backend test

* Fix lint issue

* Fix issue with node and yarn

* Fix node download

* Try github action

* Use with python

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update install_third_party_libs.py

* Update install_third_party_libs.py

* Update install_third_party_libs.py

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Make some unit test compatible with windows

* Add continue on error

* Fix linter issues

* Fix accidentely deleted line

* Fix unit tests

* Remove linter step from github action

* Fix

* Test

* Update build.yml

* Track

* Revert back

* exclude load tests

* Update github action

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Update build.yml

* Remove github action file for now.

* Enable codecoverage for Windows

* Add unit test for setup.py

* Add unit test for install_third_party_libs

* Fix lint issues

* Fix issues

Simplify return statements
Share node version and yarn version between setup and setup_test

* Fix codecoverage

* Remove redundant else statement

* Add global variable

ankita240796 committed:
Fix order (#8050)

shubha-rajan committed:
Fix #6546: Preserve GIF Animation on resize and crop (#7760)

* used gifshot and gif frames libraries to preserve animation when resizing

fixed linting errors

fixed linting errors

fixed linting errors

fixed linting errors

* added support for cropping gifs and preserving animation

* fixed linting errors

* added cursor loading view

* changed frame encoding from jpeg to png

* removed cumulative option from gif frames

Commit:
Fixes part of #6240: e2e tests for chapter create, edit, and delete (#7922)

* Tweak start script so I can access server in vm

* Add isAdmin function

Tweak createAndPublishExploration so it can handle situation when the user is admin

* Create basica chapter tests

* Add workflow for create and assign skill

* Add tests for adding and deleting skills in story editor page

* Add few more tests

* Uncomment

* Add throw for front end test

* Fix conflict in test suites

* Fix some api

* Add test for changing destination

* Fix fail to select the right topic when assign skills

* Revert tweak in start.py

* Fix linter issue

* Fix is admin

* Make code clearer

* Check warning in navigator bar

* Rename protractor class

* Fix additionalPlayerFeatures

* Rename protractor test class

aks681 committed:
fix #8024 (#8051)

ankita240796 committed:
Resolve audit issues (#8056)

ankita240796 committed:
Fix #7973: Fixes tutorial, navbar for small screens (#8044)

* Fix nav bar for small screens

* Fix tests

U8NWXD committed:
Fix Lateral Scroll Bug on Learner Dashboard Mobile (#8026)

On the learner dashboard, bootstrap by default puts padding around the
elements in rows. Since on the learner dashboard the quote image extends
across the screen from edge to edge, this padding resulted in the user
being able to scroll slightly to the right, which we do not want. To
prevent this behavior, this commit adds the `no-gutters` class to the
row, which removes the left and right padding from the row.

Fixes #7974.

vojtechjelinek committed:
Wipeout 2.9: Check snapshot content models when looking for user_id (#8036)

* Check snapshot content models when looking for user_id

* Use fetch_page to retrieve large number of models

aks681 committed:
Add editable linked skills list to question editor (#8004)

* Removed some files

* Added skill list editing to question editor

* lint

* Fixed test

* Made review changes

* Added tests

* made review changes

* made review changes

* Added 2 more object factories

* renamed variables

* fixed test

* modified comment

ankita240796 committed:
Fix modals for small screens (#8054)

aks681 committed:
Allow question linkage to skills outside topic during creation (#8045)

* Removed some files

* Added skill list editing to question editor

* lint

* Fixed test

* Made review changes

* Added tests

* Added all skill selection during question creation

* made review changes

* fixed typescript tests

* made review changes

* made review change

* Added 2 more object factories

* renamed variables

* fixed test

* modified comment

* fixed test

sagangwee committed:
Fix part of #6776: Frontend changes for listing skill opportunities on the Community Dashboard page. (#7602)

DubeySandeep committed:
Change ALL_LANGUAGE_CODES to SUPPORTED_CONTENT_LANGUAGES (#8062)

* Changes languages list variable name.

* Renamed functions.

* Edit code comments.

* Change mock variable names.

varun-tandon committed:
Fix part of #5914: Takeout Service (#7644)

* first pass: GeneralFeedbackThreadModel

* first pass GeneralFeedbackMessage

* first pass suggestion model

* add nontrivial tests

* linting

* lint

* trailing whitespace

* variable names

* resolve linting issues

* resolve linting issues

* more linting

* first pass changes

* linting

* linting

* resolve string casting issues

* switch str cast

* removed new line test

* remove spaceS

* create tests for takeout processor

* Completed activities

* incomplete data

* ExpUserLastPlaythroughModel

* LearnerPlaylistModels

* exploration ids

* StoryProgressModel

* GeneralFeedbackThreadModel

* General feedback message

* CollectionRightsData

* GeneralSuggestionModel

* ExplorationRightsModel

* Takeout test assertion

*  add future imports

* first pass takeout processor

* remove duplicate export

* reset upstream

* linting

* linting

* linting

* reiterate

* complete first version of takeout service

* added reply email model

* linting

* linting

* linting

* linting

* linting

* linting

* changes for second pass

* resolving issues for third pass

* linting

* lint

* lint

* lint

* lint

* lint

* lint

* lint

* Second pass changes

* removed unnecessary user qualifier

* resolved naming issues

* linting

* remove print statement

* remove print statement

* changes to ordering

* Added tests for trivial cases

* some linting checks

* some linting checks

* more linting fixes

* update codeowners

* resolve issue with variable naming

* add clarification for separated setup

* rearranged code

* linting

* list append rather than extend

* resolved reference issues

* linting

* linting more

tyleri committed:
Fix #8046: create command line flag for no automatic restart (#8058)

* Fix #8046: create command line flag for no automatic restart

* Fix line that was over 80 characters

seanlip committed:
Fix small error in pre-commit hook. (#8065)

DubeySandeep committed:
Fix #7994: Add frontend SkillOpportunity and ExplorationOpportunity object class (#8063)

nishantwrp committed:
Fix part of #8016:  Added a seperate backend api service for review-test-page directive  (#8061)

* Added a seperate backend api service for review-test-page directive

* Remove Linting Errors

* Remove Lint Errors Again

* Try to remove lint errors again

* Try to remove lint errors again

* Try to remove lint errors again

* Fix failing frontend_tests

* Fix failing frontend_tests again

* Add @vinitamurthi @sophiewu6 as codeowners

ankita240796 committed:
Update authors and changelog for v2.8.7 (#8070)

* Update CHANGELOG

* Update AUTHORS

* Update CONTRIBUTORS

* Update core/templates/dev/head/pages/about-page/about-page.directive.html

* Address review comments

* Fix name

* Fix name

DubeySandeep committed:
Fix issue. (#8075)

jameesjohn committed:
Fix part of #7643: Replace templateUrl with template in splash page (#7702)

* replace templateUrl with template in splash page

* use html loader

* address review comments

* removed eval

* address review comments

sagangwee committed:
Fix part of #7471: Backend changes for question suggestion integration (#7819)

vojtechjelinek committed:
Wipeout 4.1: Implement part of frontend wipeout (#8022)

* lot of cahnges

* fix linting

* fix tests

* fix frontend tests

* fix some pages

* fix webpack compilation

* fix hashes generating

* fix prod build

* fix backend tests

* fix backend test no. 256

* fix errors

* fix stuff

* add iframed error

* fix iframed error

* fix linter

* fix backend test

* fix prod build

* fix DEV_MODE

* fix backend tests

* regenerate package-lock

* assert some comments

* assert some comments

* refresh hashesa

* refresh hashesa

* ignore hashes

* codeowner

* linting

* devmode

* fix backend tests

* fix backend tests

* Fix e2e tests

* Rename webpack to webpack_bundles

* fix prod

* Fix lints

* fix e2e tests

* fix lint

* fix constatns

* lint test

* Fix requires

* Add remove account page

* Update account removal page

* wipeout frontend

* wipeout frontend fixes

* add and reorganize tests

* Address comments

* Change remove to delete

* Change js to ts

* Remove half-done changes

seanlip committed:
Revert "Fixes (partially): oppia can not not run on Windows  (#7971)" (#8087)

bansalnitish committed:
Upgrades UrlInterpolationService to Angular (#8064)

* Upgrades UrlInterpolationService to Angular

* Fixes dev_mode

* Update social-buttons.directive.ts

* Update side-navigation-bar.directive.ts

* Update top-navigation-bar.directive.ts

* Update about-page.controller.ts

* Update collection-player-page.directive.ts

* Update collection-footer.directive.ts

* Update stewards-landing-page.controller.ts

* Update learner-dashboard-page.controller.ts

* Update preferences-page.controller.ts

* Update splash-page.controller.ts

* Update story-viewer-chapters-list.directive.ts

* Update teach-page.controller.ts

* Resolves conflicts

* Resolves conflicts

* Update exploration-footer.directive.ts

* Update maintenance-page.controller.ts

* Update collection-summary-tile.directive.ts

* Update story-summary-tile.directive.ts

* Update subtopic-summary-tile.directive.ts

* Update topic-summary-tile.directive.ts

* Update learner-view-info.directive.ts

* Update conversation-skin.directive.ts

* Changes templateUrl to template

* Changes templateUrl to template

bansalnitish committed:
Upgrades Validator, ImprovementDisplay, ExplorationHTMLFormatter, SolutionObjectFactory and InteractionObjectFactory to Angular (#7989)

* Upgrades Validator service to Angular

* Upgrades ImprovementsDisplayService to Angular

* Upgrades ExplorationHtmlFormatterService to Angular

* Upgrades SolutionObjectFactory to Angular

* Upgrades InteractionObjectFactory to Angular

* Fixes conflicts

* Fixes frontend tests

* Added types

* Update improvements-tab.directive.ts

* Fix types

* Fix tests

* Fix types, test

* Fix interface name

ankita240796 committed:
Fix date formatting (#8084)

* Fix date formatting

* Add tests

* Remove extra change

* Update date-time-format.service.spec.ts

* Fix date test

Hudda committed:
Fixes part of #5939: Custom pylint extension to ensure that comments follow correct format. (#8066)

* Added lint check to ensure that comments follow correct format

* Added lint check to ensure that comments follow correct format

* Added lint check to ensure that comments follow correct format

* Changed test file for readability

Commit:
Adds log for #6921 to track sequence of steps (#8069)

* Adds sequence of steps

* fix lint issue

* add akshay as codeowner for new service; fix lint issues

* tracker.service -> debug-info-tracker.service; add specs

* remove trailing space

* address review comment

bansalnitish committed:
Upgrades StateObjectFactory and StatesObjectFactory to Angular (#8093)

* Upgrades StateObjectFactory and StatesObjectFactory to Angular

* Fixes constants

bansalnitish committed:
Upgrades TextInputRules, TextInputValidation and PencilCodeEditorRule Service to Angular (#8095)

* Upgrades TextInputRules, TextInputValidation and PencilCodeEditorRules Service to Angular

* Remove blank line

DubeySandeep committed:
Adds Ga and Aksn languages. (#8086)

jameesjohn committed:
Fix #6847: Use absolute path to require html files (#8078)

* use absolute path to require html files

* address review comments

* address review comments

* fix lint issue

* address review comments

* remove unused function

* address review comments

kevinlee12 committed:
Disable audit tests (#8104)

* Disable audit tests

* Remove unneeded function

* Remove unneeded tests

* Fix backend tests

ankita240796 committed:
Fix conflicts

ankita240796 committed:
Update tests
